File: Windows/Prosary/Localization/HebrewDisplayText.cs
namespace Prosary.Localization;

/// <summary>
/// Normalizes display-only Hebrew headings. Canonical prayer and Scripture text remains
/// pointed; titles are easier to scan without niqqud or cantillation and match the rest of the
/// app's Hebrew chrome. Hebrew punctuation such as maqaf and sof pasuq is deliberately kept.
/// </summary>
public static class HebrewDisplayText
{
    public static string WithoutMarks(string text) =>
        text.Any(IsHebrewMark) ? string.Concat(text.Where(character => !IsHebrewMark(character))) : text;

    public static string? WithoutMarksOrNull(string? text) => text is null ? null : WithoutMarks(text);

    private static bool IsHebrewMark(char character) => character is
        >= '\u0591' and <= '\u05BD' or
        '\u05BF' or
        >= '\u05C1' and <= '\u05C2' or
        >= '\u05C4' and <= '\u05C5' or
        '\u05C7' or
        '\uFB1E';
}

File: Android/app/src/main/java/com/dkaluta/prosary/typography/HebrewDisplayText.kt
package com.dkaluta.prosary.typography

/** Display-only treatment for Hebrew headings. Canonical prayer and Scripture text remains
 * pointed; headings are easier to scan without niqqud or cantillation and match the rest of the
 * app's Hebrew chrome. Hebrew punctuation (maqaf, sof pasuq, nun hafukha) is deliberately kept. */
object HebrewDisplayText {
    fun unpoint(text: String): String = buildString(text.length) {
        for (character in text) {
            if (!character.isHebrewMark()) append(character)
        }
    }

    private fun Char.isHebrewMark(): Boolean =
        code in 0x0591..0x05BD ||
            code == 0x05BF ||
            code in 0x05C1..0x05C2 ||
            code in 0x05C4..0x05C5 ||
            code == 0x05C7
}
